istioctl/cmd/describe.go

```diff
@@ -1186,3 +1199,111 @@ func containerReady(pod *v1.Pod, containerName string) (bool, error) {
+// When the client doesn't have access to root namespace, it will only show workload namespace Peerauthentications.
+func describePeerAuthentication(writer io.Writer, kubeClient kube.ExtendedClient, configClient istioclient.Interface, workloadNamespace string, podsLabels k8s_labels.Set) error { // nolint: lll
+	meshCfg, err := getMeshConfig(kubeClient)
+	if err != nil {
+		return fmt.Errorf("failed to fetch mesh config: %v", err)
+	}
+
+	workloadPAList, err := configClient.SecurityV1beta1().PeerAuthentications(workloadNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to fetch workload namespace PeerAuthentication: %v", err)
+	}
+
+	rootPAList, err := configClient.SecurityV1beta1().PeerAuthentications(meshCfg.RootNamespace).List(context.Background(), metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to fetch root namespace PeerAuthentication: %v", err)
+	}
+
+	allPAs := append(rootPAList.Items, workloadPAList.Items...)
+
+	var cfgs []*config.Config
+	for _, pa := range allPAs {
+		pa := pa
+		cfg := crdclient.TranslateObject(&pa, config.GroupVersionKind(pa.GroupVersionKind()), "")
+		cfgs = append(cfgs, &cfg)
+	}
+
+	matchedPA := findMatchedConfigs(podsLabels, cfgs)
+	printConfigs(writer, matchedPA)
+
+	activePA := authnv1beta1.ComposePeerAuthentication(meshCfg.RootNamespace, matchedPA)
+	printPeerAuthentication(writer, activePA)
+
+	return nil
+}
+
+// Workloader is used for matching all configs
+type Workloader interface {
+	GetSelector() *typev1beta1.WorkloadSelector
+}
+
+// findMatchedConfigs should filter out unrelated configs that are not matched given podsLabels.
+// When the config has no selector labels, this method will treat it as qualified namespace level
+// config. So configs passed into this method should only contains workload's namespaces configs
+// and rootNamespaces configs, caller should be responsible for controlling configs passed
+// in.
+func findMatchedConfigs(podsLabels k8s_labels.Set, configs []*config.Config) []*config.Config {
+	var cfgs []*config.Config
+
+	for _, cfg := range configs {
+		cfg := cfg
+		labels := cfg.Spec.(Workloader).GetSelector().GetMatchLabels()
+		selector := k8s_labels.SelectorFromSet(labels)
+		if selector.Matches(podsLabels) {
+			cfgs = append(cfgs, cfg)
+		}
+	}
+
+	return cfgs
+}
+
+func printConfigs(writer io.Writer, configs []*config.Config) {
+	if len(configs) == 0 {
+		fmt.Fprintf(writer, "No config found.\n")
+		return
+	}
+	fmt.Fprintf(writer, "Applied %s:\n", configs[0].Meta.GroupVersionKind.Kind)
+	var cfgNames string
+	for i, cfg := range configs {
+		cfgNames += cfg.Meta.Name + "." + cfg.Meta.Namespace
+		if i < len(configs)-1 {
+			cfgNames += ", "
+		}
+	}
+	fmt.Fprintf(writer, "   %s\n", cfgNames)
+}
+
+func printPeerAuthentication(writer io.Writer, pa *v1beta1.PeerAuthentication) {
+	fmt.Fprintf(writer, "Active PeerAuthentication:\n")
```

Comment:
Format looks like:
```
Applied PeerAuthentication:
   default.istio-system, default.default
Active PeerAuthentication:
   Workload mTLS: PERMISSIVE
```

I think it may look better like
```
Effective PeerAuthentication: PERMISSIVE
   Configs applied: default.istio-system, default.default
```
Or similar. its not clear what 'applied' and 'active' mean to me

Reply:
Updated the output accordingly.